node/src/components/contract_runtime.rs

```diff
@@ -158,18 +202,71 @@ where
     REv: From<ContractRuntimeRequest>
         + From<ContractRuntimeAnnouncement>
         + From<ControlAnnouncement>
+        + From<NetworkRequest<NodeId, Message>>
         + Send,
 {
-    type Event = ContractRuntimeRequest;
+    type Event = Event;
     type ConstructionError = ConfigError;
 
     fn handle_event(
         &mut self,
         effect_builder: EffectBuilder<REv>,
-        _rng: &mut NodeRng,
-        event: Self::Event,
+        rng: &mut NodeRng,
+        event: Event,
     ) -> Effects<Self::Event> {
         match event {
+            Event::ContractRuntimeRequest(request) => {
+                self.handle_contract_runtime_request(effect_builder, rng, *request)
+            }
+            Event::TrieRequestIncoming(request) => {
+                self.handle_trie_request(effect_builder, request)
+            }
+        }
+    }
+}
+
+impl ContractRuntime {
+    fn handle_trie_request<REv>(
```

Comment:
Since this is not a trait method, maybe a short doc comment would be nice?

Reply:
Sure, added in https://github.com/casper-network/casper-node/pull/2614/commits/9d1fd9c0687d181ac7d71c02941c19630e4e6b04.